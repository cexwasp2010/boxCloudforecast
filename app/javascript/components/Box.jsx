import React, { useState } from "react"
import './css/index.css'

import SmallBox from './images/box-small.png'
import MediumBox from './images/box-medium.png'
import LargeBox from './images/box-large.png'

import axios from 'axios';

function Box(props) {
  const [size, setSize] = useState(props.size);
  const [label, setLabel] = useState(props.label);
  const [boxOwner, setBoxOwner] = useState(props.box_owner);
  const [errorBoxOwner, setErrorBoxOwner] = useState('')
  
  const boxId = props.box_id

  const boxImage = (size) => {
    
    switch (size) {
      case 'small':
        return SmallBox
      case 'medium':
        return MediumBox
      case 'large':
        return LargeBox
    }
  }

  const handleBoxOwnerChange = (event) => {
    setBoxOwner(event.target.value);
  }

  /*Function to attach BowOwner to Box*/
  const attachBoxOwner = () => {

    /*To check box_owner is present*/
    if (!boxOwner) {
      setErrorBoxOwner("The Bow Owner is required!");
    } else {
      setErrorBoxOwner("");
      axios.post(`/attach_box_owner`, {box_owner: boxOwner, id: props.box_id})
      .then(res => {
        if (res.data.success) {
          window.location.href = '/box'
        } else {
          setErrorBoxOwner(res.data.error)
        }
      })
    }
  }

  return (
    <React.Fragment>
      <div className="card show_item">
        <div className='card-image'>
          <img src={boxImage(size)} />
        </div>
        <div className='card-content'>
          <h6><span className='brown-text'>Label: </span>{label}</h6>
          <h6><span className='brown-text'>Size: </span>{size}</h6>
          <h6>
            Attach Box Owner:
          </h6>
          <div className="input-field">
            <select value={boxOwner} onChange={handleBoxOwnerChange}>
              <option value="no_selection"></option>
              {props.box_owners && (
                props.box_owners.map((box_owner) => (
                  <option key={box_owner.value} value={box_owner.value}>{box_owner.label}</option>
                ))
              )}
            </select>
            {errorBoxOwner !== '' && (
              <div key='errorBoxOwner' className='error'>{errorBoxOwner}</div>
            )}
            <div>
              <button className='btn-small waves-effect waves-light brown lighten-1' onClick={attachBoxOwner}>Atthach box owner</button>
            </div>
          </div>
        </div>
      </div>
    </React.Fragment>
  );
}

export default Box
