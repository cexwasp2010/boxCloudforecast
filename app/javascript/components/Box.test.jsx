import React from 'react';
import { shallow } from 'enzyme';
import Box from './Box';

const label = 'Test Label';
const size ='medium'
/*Set boxOwners to construct select attach Box Owner*/
const boxOwners = [{value: 1, label: 'email1@test.com'},{value: 2, label: 'email2@test.com'}]

let wrapped = shallow(<Box label={label} size={size} box_owners={boxOwners} />);

describe('Box', () => {
  it('should render the Box Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });

  /*Fixed function to new Box.jsx*/
  it('renders the Label', () => { 
    expect(wrapped.find('h6').at(0).text()).toEqual(`Label: ${label}`);
  });

  /*Added new test to check Size Box*/
  it('renders the Size', () => { 
    expect(wrapped.find('h6').at(1).text()).toEqual(`Size: ${size}`);
  });

  /*Added new test to check Attach Box Owner field in Box*/
  it('renders the Attach Box Owner', () => { 
    expect(wrapped.find('h6').at(2).text()).toEqual(`Attach Box Owner:`);
  });

  /*Added new test to check change select attach Bow Owner into Box*/
  it('should select correct value on change', () => { 
		wrapped.find('select').simulate('change', {target: {value: 2}})
		expect(wrapped.find('select').props().value).toBe(2)
  });
});