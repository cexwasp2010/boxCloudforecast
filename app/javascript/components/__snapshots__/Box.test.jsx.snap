// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render the Box Component correctly 1`] = `
<Fragment>
  <div
    className="card show_item"
  >
    <div
      className="card-image"
    >
      <img
        src="box-medium.png"
      />
    </div>
    <div
      className="card-content"
    >
      <h6>
        <span
          className="brown-text"
        >
          Label: 
        </span>
        Test Label
      </h6>
      <h6>
        <span
          className="brown-text"
        >
          Size: 
        </span>
        medium
      </h6>
      <h6>
        Attach Box Owner:
      </h6>
      <div
        className="input-field"
      >
        <select
          onChange={[Function]}
        >
          <option
            value="no_selection"
          />
          <option
            key="1"
            value={1}
          >
            email1@test.com
          </option>
          <option
            key="2"
            value={2}
          >
            email2@test.com
          </option>
        </select>
        <div>
          <button
            className="btn-small waves-effect waves-light brown lighten-1"
            onClick={[Function]}
          >
            Atthach box owner
          </button>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
